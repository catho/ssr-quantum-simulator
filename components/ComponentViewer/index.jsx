const ComponentViewer = ({ children, name, description }) => (
  <article className="ComponentContainer">
    <h4 className="ComponentTitle">{name}</h4>
    <h6>{description}</h6>
    {children}
  </article>
);

export default ComponentViewer;
