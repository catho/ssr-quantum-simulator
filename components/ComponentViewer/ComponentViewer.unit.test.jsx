import { render, screen } from '@testing-library/react';
import ComponentViewer from '.';

const ExampleComponent = () => (
  <article>
    <p>Lorem ipsum dolor sit amet.</p>
  </article>
);

const componentMock = {
  name: 'Mock Name',
  description: 'Mock Description',
  children: <ExampleComponent />,
};

test('should match the snapshot', () => {
  const { container } = render(<ComponentViewer {...componentMock} />);
  expect(container).toMatchSnapshot();
});

test('should display the given props correctly', () => {
  render(<ComponentViewer {...componentMock} />);

  expect(screen.getByText('Mock Name')).toBeInTheDocument();
  expect(screen.getByText('Mock Description')).toBeInTheDocument();
  expect(screen.getByText('Lorem ipsum dolor sit amet.')).toBeInTheDocument();
});
