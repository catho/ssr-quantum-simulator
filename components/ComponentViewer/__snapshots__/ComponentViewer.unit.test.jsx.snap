// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <article
    class="ComponentContainer"
  >
    <h4
      class="ComponentTitle"
    >
      Mock Name
    </h4>
    <h6>
      Mock Description
    </h6>
    <article>
      <p>
        Lorem ipsum dolor sit amet.
      </p>
    </article>
  </article>
</div>
`;
